homeassistant/components/asuswrt/router.py

```diff
@@ -354,6 +372,22 @@ async def _update_unpolled_sensors(self) -> None:
             if self._sensors_data_handler.update_device_count(self._connected_devices):
                 await coordinator.async_refresh()
 
+    async def _get_available_temperature_sensors(self):
+        """Check which temperature information is available on the router."""
+        try:
+            availability = await self._api.async_find_temperature_commands()
+        except (OSError, ValueError):
+            return []
```

Comment:
Can you add test for this condition?

Reply:
Removed specific handling of `OSError` and `ValueError` since they are handled within the library.
Now response is consistent for any exception. Added a raised exception test (for `IndexError`).